File: src-tauri/src/model/result.rs
use serde::Serialize;

use crate::model::file::File;
use crate::model::letter::Letter;

#[derive(Serialize)]
pub struct Result {
    pub chars_result: Vec<Letter>,
    pub files_result: Vec<File>,
    pub duration: u64,
}

impl Result {
    pub fn new(chars_result: Vec<Letter>, files_result: Vec<File>, duration: u64) -> Self {
        Self { chars_result, files_result, duration }
    }
}

File: src/models/Result.ts
import Letter from "./Letter";
import File from "./File";

export default class Result {
    chars_result: Array<Letter>
    files_result: Array<File>
    duration: number


    constructor(chars_result: Array<Letter>, files_result: Array<File>, duration: number) {
        this.chars_result = chars_result;
        this.files_result = files_result;
        this.duration = duration;
    }


}